File: backend/src/BusOperator.Domain/Entities/Route.cs
using BusOperator.Domain.Common;

namespace BusOperator.Domain.Entities;

public class Route : AuditableEntity
{
    public int Id { get; set; }
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string FromLocation { get; set; } = string.Empty;
    public string ToLocation { get; set; } = string.Empty;
    public decimal DistanceKm { get; set; }
    public int EstimatedDurationMinutes { get; set; }
    public decimal BaseTicketPrice { get; set; }
    public bool IsActive { get; set; } = true;

    public ICollection<RouteStop> Stops { get; set; } = new List<RouteStop>();
    public ICollection<Trip> Trips { get; set; } = new List<Trip>();
}

File: frontend/types/route.ts
export interface Route {
  id: number;
  code: string;
  name: string;
  fromLocation: string;
  toLocation: string;
  distanceKm: number;
  estimatedDurationMinutes: number;
  baseTicketPrice: number;
  isActive: boolean;
}

export interface RoutePayload {
  code: string;
  name: string;
  fromLocation: string;
  toLocation: string;
  distanceKm: number;
  estimatedDurationMinutes: number;
  baseTicketPrice: number;
  isActive: boolean;
}

export interface RouteStop {
  id: number;
  routeId: number;
  stopOrder: number;
  stopName: string;
  address?: string | null;
  stopType: string;
  estimatedOffsetMinutes?: number | null;
  isActive: boolean;
}

export interface RouteStopPayload {
  stopOrder: number;
  stopName: string;
  address?: string;
  stopType: string;
  estimatedOffsetMinutes?: number | null;
  isActive: boolean;
}
